include/openPMD/Series.hpp

```diff
@@ -52,10 +52,54 @@
 namespace openPMD
 {
 class ReadIterations;
-class WriteIterations;
 class Series;
 class SeriesImpl;
 
+
+/** Writing side of the streaming API.
+ *
+ * Create instance via Series::writeIterations().
+ * For use via WriteIterations::operator[]().
+ * Designed to allow reading any kind of Series, streaming and non-
+ * streaming alike. Calling Iteration::close() manually before opening
+ * the next iteration is encouraged and will implicitly flush all
+ * deferred IO actions. Otherwise, Iteration::close() will be implicitly
+ * called upon SeriesIterator::operator++(), i.e. upon going to the next
+ * iteration in the foreach loop.
+ *
+ * Since this is designed for streaming mode, reopening an iteration is
+ * not possible once it has been closed.
+ *
+ */
+class WriteIterations : private Container< Iteration, uint64_t >
+{
+    friend class Series;
+
+private:
+    using iterations_t = Container< Iteration, uint64_t >;
+    struct SharedResources
+    {
+        iterations_t iterations;
+        auxiliary::Option< uint64_t > currentlyOpen;
+
+        SharedResources( iterations_t );
+        ~SharedResources();
+    };
+
+    using key_type = typename iterations_t::key_type;
+    using value_type = typename iterations_t::key_type;
+    WriteIterations( iterations_t );
+    explicit WriteIterations() = default;
+    //! Index of the last opened iteration
+    std::shared_ptr< SharedResources > shared;
+
+public:
+    mapped_type &
+    operator[]( key_type const & key ) override;
+    mapped_type &
+    operator[]( key_type && key ) override;
+};
+
```

Comment:
Needed to move to front now since `m_writeIterations` is now a plain member of `SeriesData`. We could use a unique pointer to avoid this.

Reply:
I think a plain member is quite good.
We could move the `WriteIterations` class in its own `.hpp`/`.cpp` files, though.